x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -1529,14 +1523,10 @@ export class RulesClient {
         await this.encryptedSavedObjectsClient.getDecryptedAsInternalUser<RawRule>('alert', id, {
           namespace: this.namespace,
         });
-      apiKeyToInvalidate = decryptedAlert.attributes.apiKey;
       attributes = decryptedAlert.attributes;
       version = decryptedAlert.version;
     } catch (e) {
-      // We'll skip invalidating the API key since we failed to load the decrypted saved object
-      this.logger.error(
-        `disable(): Failed to load API key to invalidate on alert ${id}: ${e.message}`
-      );
+      this.logger.error(`disable(): Failed to load API key of alert ${id}: ${e.message}`);
       // Still attempt to load the attributes and version using SOC
       const alert = await this.unsecuredSavedObjectsClient.get<RawRule>('alert', id);
       attributes = alert.attributes;
```

Comment:
**question:** since we no longer need any parts of the encrypted object, I wonder if we can replace the above with this fallback approach instead?

Reply:
We still decrypt the APIKey and send it back.
As i explained above, Kibana changes the encryption key (probably by using time stamp) on every update. If we don't pass apiKey and APIKeyOwner next time we want to read it Kibana cannot decrypt it and fails... Real value of APIKey remains as it is but the value in SO changes every time.

That's why i also added below in case of there is no APIKey in SO.
```
...(!attributes.apiKeyOwner && { apiKeyOwner: null }),
...(!attributes.apiKey && { apiKey: null }),
```

Reply:
Ah I see now, I will resolve this thread and move the conversation to https://github.com/elastic/kibana/pull/131581/files/b485ff8fefa3cd917cec26de1fbe9cb922e7a3ee#r869540349.